rust/src/array.rs

```diff
@@ -585,6 +585,29 @@ impl<'a> From<Vec<&'a str>> for BinaryArray {
     }
 }
 
+/// Creates a `BinaryArray` from `List<u8>` array
+impl From<ListArray> for BinaryArray {
+    fn from(v: ListArray) -> Self {
+        assert_eq!(
+            v.data().child_data()[0].child_data().len(),
+            0,
+            "BinaryArray can only be created from list array of u8 values (i.e. List<PrimitiveArray<u8>>)."
```

Comment:
@paddyhoran Do We have a test for it?

Reply:
I didn't want to be overkill on the testing as the logic is relatively simple but I'll add a few more to cover the other constraints, thanks @kszucs